### PR TITLE
Minor: Document `parquet_metadata` function

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of #7013 

## Rationale for this change

I was writing a blog post about DataFusion   https://github.com/apache/arrow-site/pull/457 and I wanted to highlight this feature that @Veeupup  added in https://github.com/apache/arrow-datafusion/issues/8367 but it wasn't documented

## What changes are included in this PR?
Document the function

## Are these changes tested?
N/A

## Are there any user-facing changes?
More docs

### CODE REVIEW

File: docs/source/user-guide/cli.md [hunk 1]
```diff
@@ -191,7 +191,7 @@ DataFusion CLI v16.0.0
 2 rows in set. Query took 0.007 seconds.
 ```
 
-## Creating external tables
+## Creating External Tables
```
Comment (alamb) [inline docs/source/user-guide/cli.md:194]: drive by cleanup -- the other headings are capitalized so it seemed strange that this one was not

File: docs/source/user-guide/cli.md [hunk 2]
```diff
@@ -467,6 +474,66 @@ Available commands inside DataFusion CLI are:
 > SET datafusion.execution.batch_size to 1024;
 ```
 
+- `parquet_metadata` table function
```
Comment (alamb) [inline docs/source/user-guide/cli.md:477]: Here is what this looks like rendered: 
Screenshot 2024-01-13 at 7 58 34 AM

File: docs/source/user-guide/cli.md [hunk 3]
```diff
@@ -425,6 +425,13 @@ Available commands inside DataFusion CLI are:
 > \h function
 ```
 
+## Supported SQL
+
+In addition to the normal [SQL supported in DataFusion], `datatfusion-cli` also
```
Comment (comphead) [inline docs/source/user-guide/cli.md:430]:
```suggestion
In addition to the normal [SQL supported in DataFusion], `datafusion-cli` also
```

File: docs/source/user-guide/cli.md [hunk 4]
```diff
@@ -467,6 +474,66 @@ Available commands inside DataFusion CLI are:
...
+| row_group_bytes         | Int64     | Number of bytes used to store the row group (not including metadata)                                |
+| column_id               | Int64     | ID of the column                                                                                    |
+| file_offset             | Int64     | Offset within the file that this column chunk's data begins                                         |
+| num_values              | Int64     | Total number of values in this column chunk                                                         |
+| path_in_schema          | Utf8      | "Path" (column name) of the column chunk in the schema                                              |
+| type                    | Utf8      | Parquet data type of the column chunk                                                               |
+| stats_min               | Utf8      | The minimum value for this column chunk, if stored in the statistics, cast to a string              |
+| stats_max               | Utf8      | The minimum value for this column chunk, if stored in the statistics, cast to a string              |
```
Comment (comphead) [inline docs/source/user-guide/cli.md:520]:
```suggestion
| stats_max               | Utf8      | The maximum value for this column chunk, if stored in the statistics, cast to a string              |
```

File: docs/source/user-guide/cli.md [hunk 5]
```diff
@@ -467,6 +474,66 @@ Available commands inside DataFusion CLI are:
...
+| stats_null_count        | Int64     | Number of null values in this column chunk, if stored in the statistics                             |
+| stats_distinct_count    | Int64     | Number of distinct values in this column chunk, if stored in the statistics                         |
+| stats_min_value         | Utf8      | Same as `stats_min`                                                                                 |
```
Comment (comphead) [inline docs/source/user-guide/cli.md:523]: wondering if this duplicated fields needed?

Comment (alamb) [reply docs/source/user-guide/cli.md:523]: I don't honestly know why the seemingly duplicated columns are present. It was done initially to mirror duckdb which has them. Maybe we should investigate the reason why 🤔 

```sql
D create table foo as select * from parquet_metadata('./benchmarks/data/hits.parquet');
D describe table foo;
┌─────────────────────────┬─────────────┬─────────┬─────────┬─────────┬─────────┐
│       column_name       │ column_type │  null   │   key   │ default │  extra  │
│         varchar         │   varchar   │ varchar │ varchar │ varchar │ varchar │
├─────────────────────────┼─────────────┼─────────┼─────────┼─────────┼─────────┤
│ file_name               │ VARCHAR     │ YES     │         │         │         │
│ row_group_id            │ BIGINT      │ YES     │         │         │         │
│ row_group_num_rows      │ BIGINT      │ YES     │         │         │         │
│ row_group_num_columns   │ BIGINT      │ YES     │         │         │         │
│ row_group_bytes         │ BIGINT      │ YES     │         │         │         │
│ column_id               │ BIGINT      │ YES     │         │         │         │
│ file_offset             │ BIGINT      │ YES     │         │         │         │
│ num_values              │ BIGINT      │ YES     │         │         │         │
│ path_in_schema          │ VARCHAR     │ YES     │         │         │         │
│ type                    │ VARCHAR     │ YES     │         │         │         │
│ stats_min               │ VARCHAR     │ YES     │         │         │         │
│ stats_max               │ VARCHAR     │ YES     │         │         │         │
│ stats_null_count        │ BIGINT      │ YES     │         │         │         │
│ stats_distinct_count    │ BIGINT      │ YES     │         │         │         │
│ stats_min_value         │ VARCHAR     │ YES     │         │         │         │
│ stats_max_value         │ VARCHAR     │ YES     │         │         │         │
│ compression             │ VARCHAR     │ YES     │         │         │         │
│ encodings               │ VARCHAR     │ YES     │         │         │         │
│ index_page_offset       │ BIGINT      │ YES     │         │         │         │
│ dictionary_page_offset  │ BIGINT      │ YES     │         │         │         │
│ data_page_offset        │ BIGINT      │ YES     │         │         │         │
│ total_compressed_size   │ BIGINT      │ YES     │         │         │         │
│ total_uncompressed_size │ BIGINT      │ YES     │         │         │         │
├─────────────────────────┴─────────────┴─────────┴─────────┴─────────┴─────────┤
│ 23 rows                                                             6 columns │
└───────────────────────────────────────────────────────────────────────────────┘
```

**Review by comphead — APPROVED** [review]
lgtm, thanks @alamb

**Pull request merged**